File: app/src/main/java/com/example/bmicalculator/MainActivity.java
package com.example.bmicalculator;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    private Button bmiButton;
    private Button ppmButton;
    private Button recipeButton;
    private  Button quizButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        bmiButton = findViewById(R.id.button_bmi);
        ppmButton = findViewById(R.id.button_ppm);
        recipeButton = findViewById(R.id.button_recipe);
        quizButton =findViewById(R.id.button_quiz);

        bmiButton.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v){
                openActivityBMI();
            }
        });
        ppmButton.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v){
                openActivityPPM();
            }
        });
        recipeButton.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v){
                openActivityRecipes();
            }
        });
        quizButton.setOnClickListener(v -> openActivityQuiz());


    }

    public void openActivityBMI(){
        Intent intent = new Intent(this, BMI.class);
        startActivity(intent);
    }

    public void openActivityPPM(){
        Intent intent = new Intent(this, PPM.class);
        startActivity(intent);
    }

    public void openActivityRecipes(){
        Intent intent = new Intent(this, Recipe.class);
        startActivity(intent);
    }
    public void openActivityQuiz(){
        Intent intent = new Intent(this, Quiz.class);
        startActivity(intent);
    }

}

File: app/src/main/java/com/example/bmicalculator/MainActivity.kt
package com.example.bmicalculator

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        button_bmi.setOnClickListener { openActivityBMI() }
        button_ppm.setOnClickListener { openActivityPPM() }
        button_recipe.setOnClickListener { openActivityRecipes() }
        button_quiz.setOnClickListener { openActivityQuiz() }
    }

    private fun openActivityBMI() {
        val intent = Intent(this, BMI::class.java)
        startActivity(intent)
    }

    private fun openActivityPPM() {
        val intent = Intent(this, PPM::class.java)
        startActivity(intent)
    }

    private fun openActivityRecipes() {
        val intent = Intent(this, Recipe::class.java)
        startActivity(intent)
    }

    private fun openActivityQuiz() {
        val intent = Intent(this, Quiz::class.java)
        startActivity(intent)
    }
}